types/index.d.ts

```diff
@@ -0,0 +1,154 @@
+  export import MoneroMultisigSignResult = MoneroMultisigSignResult;
+  import * as MoneroOutputWallet from "./src/main/js/wallet/model/MoneroOutputWallet";
+  export import MoneroOutputWallet = MoneroOutputWallet;
+  import * as MoneroOutputQuery from "./src/main/js/wallet/model/MoneroOutputQuery";
+  export import MoneroOutputQuery = MoneroOutputQuery;
+  import * as MoneroTxPriority from "./src/main/js/wallet/model/MoneroTxPriority";
+  export import MoneroTxPriority = MoneroTxPriority;
+  import * as MoneroTxConfig from "./src/main/js/wallet/model/MoneroTxConfig";
+  export import MoneroTxConfig = MoneroTxConfig;
+  import * as MoneroSubaddress from "./src/main/js/wallet/model/MoneroSubaddress";
+  export import MoneroSubaddress = MoneroSubaddress;
+  import * as MoneroSyncResult from "./src/main/js/wallet/model/MoneroSyncResult";
+  export import MoneroSyncResult = MoneroSyncResult;
+  import * as MoneroTransfer from "./src/main/js/wallet/model/MoneroTransfer";
+  export import MoneroTransfer = MoneroTransfer;
+  import * as MoneroIncomingTransfer from "./src/main/js/wallet/model/MoneroIncomingTransfer";
+  export import MoneroIncomingTransfer = MoneroIncomingTransfer;
+  import * as MoneroOutgoingTransfer from "./src/main/js/wallet/model/MoneroOutgoingTransfer";
+  export import MoneroOutgoingTransfer = MoneroOutgoingTransfer;
+  import * as MoneroTransferQuery from "./src/main/js/wallet/model/MoneroTransferQuery";
+  export import MoneroTransferQuery = MoneroTransferQuery;
+  import * as MoneroTxSet from "./src/main/js/wallet/model/MoneroTxSet";
+  export import MoneroTxSet = MoneroTxSet;
+  import * as MoneroTxWallet from "./src/main/js/wallet/model/MoneroTxWallet";
+  export import MoneroTxWallet = MoneroTxWallet;
+  import * as MoneroTxQuery from "./src/main/js/wallet/model/MoneroTxQuery";
+  export import MoneroTxQuery = MoneroTxQuery;
+  import * as MoneroWalletListener from "./src/main/js/wallet/model/MoneroWalletListener";
+  export import MoneroWalletListener = MoneroWalletListener;
+  import * as MoneroWalletConfig from "./src/main/js/wallet/model/MoneroWalletConfig";
+  export import MoneroWalletConfig = MoneroWalletConfig;
+  import * as MoneroMessageSignatureType from "./src/main/js/wallet/model/MoneroMessageSignatureType";
+  export import MoneroMessageSignatureType = MoneroMessageSignatureType;
+  import * as MoneroMessageSignatureResult from "./src/main/js/wallet/model/MoneroMessageSignatureResult";
+  export import MoneroMessageSignatureResult = MoneroMessageSignatureResult;
+  import * as MoneroConnectionManager from "./src/main/js/common/MoneroConnectionManager";
+  export import MoneroConnectionManager = MoneroConnectionManager;
+  import * as MoneroConnectionManagerListener from "./src/main/js/common/MoneroConnectionManagerListener";
+  export import MoneroConnectionManagerListener = MoneroConnectionManagerListener;
+  import * as MoneroUtils from "./src/main/js/common/MoneroUtils";
```

Comment:
I think the import and export lines for all of these things could be replaced with something like:
`export * as MoneroUtils from "monero-javascript/types/src/main/js/common/MoneroUtils";` .  The index file is pulling all of these declaration files into one place and re-exporting;  Note that I've replaced the actual JS location with the type location- this is important.  Try importing your branch into a typescript project and call `const utils = monerojs.MoneroUtils` and you'll see that the definitions and methods now pop up.

Reply:
Additionally, this index.d.ts file location should (I think) be referenced in the monero-javascript package.json as `"types": "types/index.d.ts"`